gbpservice/nfp/configurator/drivers/firewall/asav/asav_fw_driver.py

```diff
@@ -0,0 +1,960 @@
+                elif iptools.ipv4.validate_cidr(
+                        rule["destination_ip_address"]):
+                    destinationAddress = dict(
+                        kind="IPv4Network",
+                        value=rule["destination_ip_address"])
+                elif iptools.ipv4.validate_ip(rule["destination_ip_address"]):
+                    destinationAddress = dict(
+                        kind="IPv4Address",
+                        value=rule["destination_ip_address"])
+
+            if rule["source_ip_address"]:
+                if rule["source_ip_address"] == const.REFRENCE_IPS['WILDCARD']:
+                    sourceAddress = dict(kind="AnyIPAddress",
+                                         value=rule["source_ip_address"])
+                elif iptools.ipv4.validate_cidr(rule["source_ip_address"]):
+                    sourceAddress = dict(kind="IPv4Network",
+                                         value=rule["source_ip_address"])
+                elif iptools.ipv4.validate_ip(rule["source_ip_address"]):
+                    sourceAddress = dict(kind="IPv4Address",
+                                         value=rule["source_ip_address"])
+            else:
+                sourceAddress = dict(kind="AnyIPAddress",
+                                     value=const.REFRENCE_IPS['WILDCARD'])
+
+            # Supported protocol: TCP, UDP and ICMP.
+            # Agent doesn't do any validation of protocol and relies on plugin
+            # or in case of Sungaard on UI to do validation.
+            if source_port:
+                if ':' in source_port:
+                    source_port = '-'.join(source_port.split(':'))
+                src_value = rule['protocol'] + "/" + str(source_port)
+                sourceService = dict(kind="TcpUdpService",
+                                          value=src_value)
+            elif not source_port:
+                sourceService = dict(kind="NetworkProtocol",
+                                     value=rule["protocol"])
+
+            if destination_port:
+                if ':' in destination_port:
+                    destination_port = '-'.join(destination_port.split(':'))
+                dst_value = rule['protocol'] + "/" + str(destination_port)
+                destinationService = dict(kind="TcpUdpService",
+                                          value=dst_value)
+            elif not destination_port:
+                destinationService = dict(kind="NetworkProtocol",
+                                          value=rule['protocol'])
+
+            if rule["action"] == "allow":
+                permit = True
+            else:
+                permit = False
+            asav_access_rule = dict(
+                position=rule['position'],
+                permit=permit
+            )
+
+            try:
+                asav_access_rule.update(
+                    {'destinationAddress': destinationAddress})
+            except NameError:
```

Comment:
We are continuing further even if the rule update fails. Will there be datapath issue because of this?